cranelift/interpreter/src/step.rs

```diff
@@ -668,11 +668,40 @@ where
         Opcode::IshlImm => binary(Value::shl, arg(0)?, imm_as_ctrl_ty()?)?,
         Opcode::UshrImm => binary(Value::ushr, arg(0)?, imm_as_ctrl_ty()?)?,
         Opcode::SshrImm => binary(Value::ishr, arg(0)?, imm_as_ctrl_ty()?)?,
-        Opcode::Bitrev => unimplemented!("Bitrev"),
-        Opcode::Clz => unimplemented!("Clz"),
-        Opcode::Cls => unimplemented!("Cls"),
-        Opcode::Ctz => unimplemented!("Ctz"),
-        Opcode::Popcnt => unimplemented!("Popcnt"),
+        Opcode::Bitrev => assign(Value::reverse_bits(arg(0)?)?),
+        // For `Clz`, `Cls`, `Ctz`, and `Popcnt`, the underlying Rust function
+        // always returns `u32` (and therefore a `Value` of type `U32`), so this
+        // is switched back to the correct type by recreating the `Value`.
+        Opcode::Clz => assign(Value::int(
+            Value::leading_zeros(arg(0)?)?.into_int()?,
+            ctrl_ty,
+        )?),
```

Comment:
Would it make sense for us to instead do a `.convert(ValueConversionKind::Exact(ctrl_ty))`, instead of casting to and from int?

```suggestion
        Opcode::Clz => assign(
            arg(0)?
                .leading_zeroes()?
                .convert(ValueConversionKind::Exact(ctrl_ty))?
         )?,
```

I think Exact would be able to both extend and reduce types? Although I'm not sure about converting from unsigned to signed types.

Reply:
Or we could cast it to the original type in the macro that `leading_zeroes` uses, so that it would always return the original type? I think that would also solve a lot of issues and make the rest of the interface cleaner.